### DISCUSSION TITLE
Mobile (random) SAM batteries and defined number of aircraft per squadron.

### DISCUSSION THREAD

**Discussion by Froggy94** [Khopa/dcs_liberation#2151] [Q&A] [answered]
Hello, 

First of all, thank you very much for your excellent work. I tried to create a campaign based on the Bosnian war. There was a big threat from mobile Sam batteries. Normally the batteries are not always in one place (otherwise they are easy targets). Therefore I think it makes sense to implement a system for dynamic placement of SAM batteries in Liberation. In general, I would suggest allocating a contingent of SAM batteries to the AI faction.  A random number of batteries from this contingent would then be randomly distributed to predefined points when generating missions. These points would have to be set in the editor when creating the campaign. Together with Skynet, a highly dynamic and unpredictable threat situation could be created.  Here is some simple code, how I imagine the distribution of the SAM batteries (possibly not error-free):

    # Method takes a List of available SAM Batteries and possible SAM Sites and returns
    # a list with active Sam Sites To which a SAM battery was added
    def generateMoblieSAMSites(allAvailableSAMBatteries, allpossibleSAMSites):

        maxNumberofDepolyedSamBatteries  = len(allAvailableSAMBatteries)

        if len(allpossibleSAMSites) < maxNumberofDepolyedSamBatteries:
            maxNumberofDepolyedSamBatteries = len(allpossibleSAMSites)

        numberOfDeployedSAMBatteries = random.randint(0, maxNumberofDepolyedSamBatteries)

        currentAvailableSAMBatteries = allAvailableSAMBatteries.copy()
        currentPossibleSAMSites = allpossibleSAMSites.copy()

        activeSAMSites = []

        for i in range(numberOfDeployedSAMBatteries):

            indexOfBatterie = random.randint(0, len(currentAvailableSAMBatteries))
            indexOfSite =  random.randint(0, len(currentPossibleSAMSites))

            batterie = currentAvailableSAMBatteries.pop(indexOfBatterie)
            site = currentPossibleSAMSites.pop(indexOfSite)

            site.addBatterie(batterie)
            activeSAMSites.append(site)

        return activeSAMSites

I also think it can be useful to define the number of aircraft per squadron for the AI faction. The AI then tries to buy exactly the number of aircraft. This should make it possible to define the capabilities of the AI faction more precisely in advance. In addition, it should be defined whether the AI buys more aircraft for the squadron if they are lost. I think this can be defined via the squadron configuration:

   - primary: TARCAP
      secondary: any
      aircraft:
        - MiG-29A Fulcrum-A
       amount
        - 12
       reinforce
        - True

**Comment by Starfire13** [comment, marked as answer]
https://github.com/dcs-liberation/dcs_liberation/issues/538
https://github.com/dcs-liberation/dcs_liberation/issues/1594
